includes/classes/class-fixinsecurecontent.php

```diff
@@ -0,0 +1,313 @@
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			// translators: Message to show when the fixing of insecure content is completed.
+			$message = PHP_EOL . sprintf( __( 'Total posts checked for insecure URL(s): %s', 'insecure-content-warning' ), $this->total_post_count ) . PHP_EOL;
+			WP_CLI::log( WP_CLI::colorize( "%c{$message}%n " ) );
+			Utils\format_items( 'table', $this->fixed_post_count, array( 'URL(s) fixed summary' ) );
+		} else {
+			if ( null === $this->fixed_post_count ) {
+				return __( 'No post(s) found', 'insecure-content-warning' );
+			}
+
+			$output = '';
+
+			foreach ( $this->warning_count as $warning ) {
+				$warning_message = array_shift( $warning );
+				$output         .= '<span class="warning">' . __( 'Warning', 'insecure-content-warning' ) . '</span>: ' . $warning_message . PHP_EOL;
+			}
+
+			foreach ( $this->fixed_post_count as $fixed_post ) {
+				$output .= array_shift( $fixed_post ) . PHP_EOL;
+			}
+
+			return $output;
+		}
+	}
+
+	/**
+	 * Fix insecure content for given Post ID.
+	 *
+	 * @param int $post_id Post ID.
+	 *
+	 * @return array|string
+	 */
+	protected function fix_insecure_content( $post_id ) {
+		$current_post = get_post( $post_id );
+
+		// Check and make sure post exists.
+		if ( empty( $current_post ) ) {
+			// translators: Message to show when the system is unable to fetch details for the post.
+			$message = sprintf( __( 'Unable to fetch details for post %s', 'insecure-content-warning' ), $post_id );
+			if ( defined( 'WP_CLI' ) && WP_CLI ) {
+				// translators: Message to show when the system is unable to fetch details for the post.
+				WP_CLI::warning( $message );
+			} else {
+				$this->warning_count[] = array( __( 'Warning', 'insecure-content-warning' ) => $message );
+			}
+			return '';
+		}
+
+		// Loop through post content and get HTTPS URLs wherever possible.
+		$post_content = $current_post->post_content;
+		$urls         = $this->parse_content_for_insecure_urls( $post_content );
+		$count        = 0;
+
+		// Check if post content has insecure URLs.
+		if ( empty( $urls ) ) {
+			// translators: Message to show when no insecure content URL found in the post.
+			$message = sprintf( __( 'No insecure content URL found in post %s', 'insecure-content-warning' ), $post_id );
+			if ( defined( 'WP_CLI' ) && WP_CLI ) {
+				WP_CLI::warning( $message );
+			} else {
+				$this->warning_count[] = array( __( 'Warning', 'insecure-content-warning' ) => $message );
+			}
+
+			if ( $this->dry_run ) {
+				// translators: Summary to show on dry run fix.
+				$this->fixed_post_count[] = array( __( 'URL(s) fixed summary', 'insecure-content-warning' ) => sprintf( __( '0/0 URL(s) will be fixed in post %s', 'insecure-content-warning' ), $post_id ) );
+			} else {
+				// translators: Summary to show on fix.
+				$this->fixed_post_count[] = array( __( 'URL(s) fixed summary', 'insecure-content-warning' ) => sprintf( __( '0/0 URL(s) fixed in post %s', 'insecure-content-warning' ), $post_id ) );
```

Comment:
As above re translator note and use of `%d`

Reply:
Done, thank you.